blockchain/treasury.go

```diff
@@ -519,7 +519,7 @@ type TreasuryBalanceInfo struct {
 // TreasuryBalance returns treasury balance information as of the given block.
 func (b *BlockChain) TreasuryBalance(hash *chainhash.Hash) (*TreasuryBalanceInfo, error) {
 	node := b.index.LookupNode(hash)
-	if node == nil || !b.index.NodeStatus(node).HaveData() {
+	if node == nil || !b.index.CanValidate(node) {
```

Comment:
Right.  Previously, the only way `HaveData` would be true is also if all of the ancestors also had their data and, in most cases, the block had been processed.  However, there was also the case where a side chain block that was never fully validated would pass through the old `HaveData` check and fail below as well.

So, I believe they're actually the same in that regard.  That is to say that blocks that haven't been processed yet will fail in the same way it already used to for side chain blocks that hadn't been processed because they were never connected.

I concur that it isn't really a problem either way because the only time anyone should realistically be querying this is once a block is known to be part of the best chain (e.g. connected).